go/exec/exec.go

```diff
@@ -33,21 +44,78 @@ type Exec struct {
 	AnsibleConfig ansible.Config
 	Infra         infra.Infra
 
+	// rootDir represents the parent directory of the Exec.
+	// From there, the Exec's unique directory named Exec.dirPath will
+	// be created once Exec.Prepare is called.
 	rootDir string
+
+	// dirPath is Exec's unique directory where all reports, directories,
+	// files, and logs are kept.
 	dirPath string
+
+	stdout io.Writer
+	stderr io.Writer
+
+	prepared bool
 }
 
+// SetStdout sets the standard output of Exec.
+func (e *Exec) SetStdout(stdout *os.File) {
+	e.stdout = stdout
+	e.AnsibleConfig.SetStdout(stdout)
+}
+
+// SetStderr sets the standard error output of Exec.
+func (e *Exec) SetStderr(stderr *os.File) {
+	e.stderr = stderr
+	e.AnsibleConfig.SetStderr(stderr)
+}
+
+// SetOutputToDefaultPath sets Exec's outputs to their default files (stdoutFile and
+// stderrFile). If they can't be found in Exec.dirPath, they will be created.
+func (e *Exec) SetOutputToDefaultPath() error {
+	if !e.prepared {
+		return errors.New(ErrorNotPrepared)
+	}
+	outFile, err := os.OpenFile(path.Join(e.dirPath, stdoutFile), os.O_RDWR | os.O_CREATE, 0755)
+	if err != nil {
+		return err
+	}
+
+	errFile, err := os.OpenFile(path.Join(e.dirPath, stderrFile), os.O_RDWR | os.O_CREATE, 0755)
+	if err != nil {
+		return err
+	}
+
+	e.stdout = outFile
+	e.stderr = errFile
+	e.AnsibleConfig.SetOutputs(outFile, errFile)
+	return nil
+}
+
+// Prepare prepares the Exec for a future Execution.
 func (e *Exec) Prepare() error {
 	err := e.prepareDirectories()
 	if err != nil {
 		return err
 	}
 
+	err = e.Infra.Prepare()
+	if err != nil {
+		return err
+	}
+	e.prepared = true
```

Comment:
should this be checked so we don't accidentally prepare an Exec twice?

Reply:
Good finding! Fixed in a80ed41e4e86a7c54a6e4e4a73a6028ab3482942.